src/subscription_manager/i18n.py

```diff
@@ -45,32 +45,40 @@ def configure_i18n():
     except locale.Error:
         # Following message could be little bit confusing. Why? When environment variable
         # LANG is set to e.g. es_ES.UTF-8, then we can show localized message for this language,
-        # but this language could not be fully supported by the system. It means that en_ES is not
+        # but this language could not be fully supported by the system. It means that es_ES is not
         # listed in the list of 'locale -a'
-        _locale = None
-        if "LANG" in os.environ:
-            _locale = os.environ["LANG"]
-        elif "LC_ALL" in os.environ:
-            _locale = os.environ["LC_ALL"]
+
+        # Determine the locale that setlocale attempted to set to use in the warning message.
+        # There is a cornercase if the env var that setlocale() raised an error for is neither
+        # LC_ALL nor LANG but we ignore that for now.
+        _locale = os.environ.get("LC_ALL", None)
+        if not _locale:
+            _locale = os.environ.get("LANG", None)
+
         if _locale is not None and Locale.is_locale_supported(_locale):
             print(
                 'You are attempting to use a locale: "%s" that is not fully supported by this system.'
                 % _locale
             )
+        ### Is this fine?  C.UTF-8 doesn't exist on RHEL7 so these can lead to exceptions.
         os.environ["LC_ALL"] = "C.UTF-8"
         locale.setlocale(locale.LC_ALL, "C.UTF-8")
 
     configure_gettext()
     # RHBZ 1642271  Don't set a None lang
-    lang = os.environ.get("LANG")
+    lang = locale.setlocale(locale.LC_MESSAGES, None)
```

Comment:
Couldn't this raise exception? The exception is not handled in this case. What is wrong with `lang = os.environ.get("LANG")`?

Reply:
I believe that after we get through the try except block above, this can no longer raise an exception.

setting lang equal to the value of LANG is bad because it doesn't obey any of the other environment variables.  Of the environment variables that control translations, LANG is the fallback value.  All of the other environment variables have higher prefence and therefore if any of them are set, they should be used instead of LANG.

`setlocale(LC_MESSAGES)` used at this point, after the call to `setlocale(LC_ALL, "")`, should return the proper value for translations taking into account all of the relevant environment variables.  The other option would be to code our own set of fallbacks that follow the standards, examining LANGUAGE, LC_ALL, LC_MESSAGES, and LANG for the relevant language code to use for translations.